### ISSUE TITLE
Auto RTL support

### ISSUE THREAD

**Issue by themedleb** [Dullage/flatnotes#43]
Like adding this HTML attribute `dir="auto"` to the nodes to make them change direction automatically based on the typed character.

Right now, I'm using hackmd.io they don't support auto RTL support though, but at least they support YAML so I can use something like:

```
---
dir: rtl
---
```
to make the whole page RTL when needed.

I wish I can help but I only know some basic HTML and CSS :sweat_smile:

**Comment by Dullage** [comment]
This isn't something I'm familiar with but, if it's as simple as it looks, then I'm sure it's something that could be added. I'll take a look.

**Comment by themedleb** [comment]
Thank you, very appreciated.

**Comment by Dullage** [comment]
I had a play around with this attribute, adding it to a number of different nodes. Unfortunately it didn't seem to have any effect. I'll flag this as 'Help Needed' and perhaps someone with more experience in this area can lend a hand.

**Comment by themedleb** [comment]
It's fine, I think it has to do with the tiptap project.

Either way, thank you for trying :heart: .

**Comment by pbogre** [comment]
> I had a play around with this attribute, adding it to a number of different nodes. Unfortunately it didn't seem to have any effect. I'll flag this as 'Help Needed' and perhaps someone with more experience in this area can lend a hand.

i believe it only takes effect for arabic characters, I was able to see it change the direction of a title written in arabic characters

**Comment by dullage** [comment]
@pbogre - Thanks for having a stab at this. I can see now that this property is changing the input direction (e.g. the 'Home' key takes you to the right of the text rather than the left) but it isn't changing the text alignment (i.e. the text is still aligned left). 

I don't know this for sure as I've never dealt with rtl languages but I can imagine that the alignment is as important as the input direction and so it's giving me pause on merging this change. Any thoughts?

**Comment by pbogre** [comment]
You're absolutely right i don't know how I didn't think about that. Honestly I'm not familiar with RTL languages either so I couldn't tell you how important it is to have the alignment be on the right and not just the directionality.

Perhaps someone with more knowledge on the matter could give us some insight, as for aligning the actual text to the right when the direction is rtl i'm not sure how to implement that the `dir=auto` attribute is applied to the whole document and not just the elements that require it, since in that case it would just be a matter of also adding `float: right` to the style.

If someone can come up with a better solution for this then that would definitely solve this, but as of now I think proper directionality is the very least amount of support that flatnotes should offer for RTL languages
